src/Protocol/Diaspora/Factory/DiasporaContact.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Friendica\Protocol\Diaspora\Factory;
+
+use Friendica\Capabilities\ICanCreateFromTableRow;
+use Friendica\Database\DBA;
+use GuzzleHttp\Psr7\Uri;
+
+class DiasporaContact extends \Friendica\BaseFactory implements ICanCreateFromTableRow
+{
+	public function createFromTableRow(array $row): \Friendica\Protocol\Diaspora\Entity\DiasporaContact
+	{
+		return new \Friendica\Protocol\Diaspora\Entity\DiasporaContact(
+			new Uri($row['url']),
+			$row['uri-id'],
+			$row['guid'],
+			$row['addr'],
+			$row['alias'] ? new Uri($row['alias']) : null,
+			$row['nick'],
+			$row['name'],
+			$row['given-name'],
+			$row['family-name'],
+			$row['alias'] ? new Uri($row['photo']) : null,
```

Comment:
Why is the "photo" field depending on the "alias" field here and in the following lines?

Reply:
Good catch, it doesn't. Copy-paste issue.